src/components/MapVizzard/MapLayers/MapLayers.jsx

```diff
@@ -0,0 +1,63 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import styles from "./MapLayers.module.css";
+import LayerRow from "./LayerRow";
+
+export const MapLayers = ({
+  layers,
+  setLayers,
+  val,
+  setVal,
+  activeLayer,
+  setActiveLayer,
+}) => {
+  const renderLayers = [];
+
+  const updateLayers = (d, id) => {
+    let l = layers.filter((dd) => dd.id == id);
+    if (d == "remove") {
+      l.forEach((f) =>
+        layers.splice(layers.findIndex((e) => e.id === f.id),1)
+      );
+    } else {
+      l = d;
+    }
+    if(setLayers) setLayers([...layers]);
+  };
+
+  const mLayers = layers.sort(function (a, b) {
+    return b["zIndex"] - a["zIndex"];
+  });
```

Comment:
Let's not use sort as this will mutate the input.
Let's make a copy before we sort an argument